src/analysis.rs

```diff
@@ -209,13 +209,19 @@ impl Analysis {
         self.for_each_crate(|c| c.defs_per_file.get(file).map(&f))
     }
 
-    pub fn defs_for_name(&self, name: &str) -> Vec<Def> {
+    pub fn defs_matching_query(&self, query: &str) -> Vec<Def> {
+        let lowered_query = query.to_lowercase();
+
         self.for_all_crates(|c| {
-            c.def_names.get(name).map(|ids| {
-                ids.into_iter()
-                    .flat_map(|id| c.defs.get(id).cloned())
-                    .collect()
-            })
+            let defs: Vec<Def> = c.def_names.iter().fold(vec![], |mut acc, (ref def_n, ref ids)| {
```

Comment:
I would use filter and flat_map, rather than fold, I think the code would be simpler and more efficient.